src/main/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollection.java

```diff
@@ -151,67 +169,98 @@ public TraversalParameters getTraversalParameters( final SAMSequenceDictionary s
             throw new GATKException("Cannot call getTraversalParameters() without specifying either intervals to include or exclude.");
         }
 
+        if (intervalMergingRule == IntervalMergingRule.NONE) {
+            logger.warn("The user has specified `--interval-merging-rule NONE`. This is an advanced traversal mode that may cause problems with engine traversal code. Use at your own risk.");
+        }
+
         if ( traversalParameters == null ) {
-            parseIntervals(new GenomeLocParser(sequenceDict));
+            traversalParameters = parseIntervals(new GenomeLocParser(sequenceDict), intervalMergingRule, intervalSetRule, excludeIntervalStrings);
         }
 
         return traversalParameters;
     }
 
-    private void parseIntervals(final GenomeLocParser genomeLocParser) {
+    private TraversalParameters parseIntervals(final GenomeLocParser genomeLocParser, final IntervalMergingRule intervalMergingRule,
+                                               final IntervalSetRule intervalSetRule, final List<String> excludeIntervalStrings) {
         // return if no interval arguments at all
         if (!intervalsSpecified()) {
             throw new GATKException("Cannot call parseIntervals() without specifying either intervals to include or exclude.");
         }
 
-        GenomeLocSortedSet includeSortedSet;
-        if (getIntervalStrings().isEmpty()){
-            // the -L argument isn't specified, which means that -XL was, since we checked intervalsSpecified()
-            // therefore we set the include set to be the entire reference territory
-            includeSortedSet = GenomeLocSortedSet.createSetFromSequenceDictionary(genomeLocParser.getSequenceDictionary());
+        if (intervalMergingRule == IntervalMergingRule.NONE ) {
+            if ( !excludeIntervalStrings.isEmpty()) {
+                throw new UserException("-XL and '--interval-merging-rule NONE' is a currently unsupported argument combination");
+            }
+            if ( intervalSetRule != IntervalSetRule.UNION) {
+                throw new UserException("'--"+intervalSetRule+"'and '--interval-merging-rule NONE' is a currently unsupported argument combination");
```

Comment:
`"'--"+intervalSetRule+"` -> `"'--interval-set-rule " + intervalSetRule + "' and...`